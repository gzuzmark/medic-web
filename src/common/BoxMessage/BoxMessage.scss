.BoxMessage {
  align-items: center;
  border-radius: 8px;
  color: white;
  display: flex;
  height: 40px;
  padding: 0 24px;
  &--error {
    background: #ee5c7d;
  }
}