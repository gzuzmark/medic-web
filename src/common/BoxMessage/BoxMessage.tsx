// TODO: Hola Carlos del futuro esto quizas te sirva para mejorar el buscador de la lista => https://react-select.com/home#async

import * as React from 'react';
import { Text } from '../ConsoleText';
import Icon from "../Icon/Icon";
import './BoxMessage.scss';


interface IPropsBoxMessage {
    type: string;
    show: boolean;
}

const getKeyIcon = (key: string): string => {
    let icon = '';
    if (key === 'error') {
        icon = 'exclamation'
    }
    return icon;
};

const BoxMessage: React.StatelessComponent<IPropsBoxMessage> = (props) => {
    if (!props.show) {
        return null;
    }
    return (
        <div className={`BoxMessage BoxMessage--${props.type}`}>
            <Icon name={getKeyIcon(props.type)} style={{fill: 'white', paddingRight: 10, height: 30, width: 30}}/>
            <Text color={'LightText'}>{props.children}</Text>
        </div>
    );
};

export default BoxMessage;