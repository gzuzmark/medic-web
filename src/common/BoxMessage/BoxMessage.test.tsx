import { shallow } from 'enzyme';
import 'jest-localstorage-mock';
import * as React from 'react';
import BoxMessage from './BoxMessage';

describe('BoxMessage Test',() => {
    let props: any;
    let mountedBoxMessage: any;
    const getComponent = () => {
        if (!mountedBoxMessage) {
            mountedBoxMessage = shallow(
                <BoxMessage {...props}>Caja de Mensajes</BoxMessage>
            );
        }
        return mountedBoxMessage;
    };

    beforeEach(() => {
        props = {
            show: true,
            type: ''
        };
        mountedBoxMessage = undefined;
    });

    it("render: render BoxMessage", () => {
        props = {
            show: true,
            type: 'error'
        };
        const component = getComponent();
        expect(component).toMatchSnapshot();
    });

    it("render: render BoxMessage should be null", () => {
        props = {
            show: false,
            type: 'error'
        };
        const component = getComponent().find('.BoxMessage');
        expect(component.length).toBe(0);
    });

});