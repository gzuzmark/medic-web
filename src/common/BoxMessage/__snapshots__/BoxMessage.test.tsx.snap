// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxMessage Test render: render BoxMessage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BoxMessage
    show={true}
    type="error"
  >
    Caja de Mensajes
  </BoxMessage>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          name="exclamation"
          style={
            Object {
              "fill": "white",
              "height": 30,
              "paddingRight": 10,
              "width": 30,
            }
          }
        />,
        <styled.span
          color="LightText"
        >
          Caja de Mensajes
        </styled.span>,
      ],
      "className": "BoxMessage BoxMessage--error",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "exclamation",
          "style": Object {
            "fill": "white",
            "height": 30,
            "paddingRight": 10,
            "width": 30,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Caja de Mensajes",
          "color": "LightText",
        },
        "ref": null,
        "rendered": "Caja de Mensajes",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            name="exclamation"
            style={
              Object {
                "fill": "white",
                "height": 30,
                "paddingRight": 10,
                "width": 30,
              }
            }
          />,
          <styled.span
            color="LightText"
          >
            Caja de Mensajes
          </styled.span>,
        ],
        "className": "BoxMessage BoxMessage--error",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "exclamation",
            "style": Object {
              "fill": "white",
              "height": 30,
              "paddingRight": 10,
              "width": 30,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Caja de Mensajes",
            "color": "LightText",
          },
          "ref": null,
          "rendered": "Caja de Mensajes",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
